python/lib/physiological.py

```diff
@@ -520,19 +522,27 @@ def insert_event_file(self, event_data, event_file, physiological_file_id,
             for field in optional_fields:
                 if field not in row.keys():
                     row[field] = None
+
             # TODO: remove the following if once received confirmation from
             # TODO: James it was an error.
             if "NaN" in row['duration']:
                 row['duration'] = 0
+
+            sample = None
+            if row['event_sample'] and (type(row['event_sample']) == int or type(row['event_sample']) == float):
+                sample = row['event_sample']
+            if row['sample'] and (type(row['sample']) == int or type(row['sample']) == float):
+                sample = row['sample']
+
             values_tuple = (
                 str(physiological_file_id),
                 row['onset'],
                 row['duration'],
                 row['trial_type'],
                 row['response_time'],
                 row['event_code'],
-                row['event_value']  or row['value'],
-                row['event_sample'] or row['sample'],
+                str(row['event_value']) or str(row['value']),
```

Comment:
this is causing an error for datasets that do not have an event_value column in the events file (example: Face13 dataset that has only onset, duration and trial_type set in the TSV file). 

So in this case, row['event_value'] and row['value'] are set to `None` and when you convert it to `str` it tries to insert the string "None" in the `EventValue` field that should be a 10 digits integer causing the following failure:

```
Traceback (most recent call last):
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/database.py", line 185, in insert
    cursor.executemany(query, values)
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 230, in executemany
    return self._do_execute_many(
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 261, in _do_execute_many
    rows += self.execute(sql + postfix)
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 206, in execute
    res = self._query(query)
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 319, in _query
    db.query(q)
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/connections.py", line 259, in query
    _mysql.connection.query(self, query)
MySQLdb._exceptions.OperationalError: (1366, "Incorrect integer value: 'None' for column `loris_rb`.`physiological_task_event`.`EventValue` at row 1")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cmadjar/Development/Github/loris-mri/python/bids_import.py", line 463, in <module>
    main()
  File "/Users/cmadjar/Development/Github/loris-mri/python/bids_import.py", line 76, in main
    read_and_insert_bids(bids_dir, config_file, verbose, createcand, createvisit)
  File "/Users/cmadjar/Development/Github/loris-mri/python/bids_import.py", line 220, in read_and_insert_bids
    Eeg(
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/eeg.py", line 153, in __init__
    self.register_data()
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/eeg.py", line 260, in register_data
    event_file_path     = self.fetch_and_insert_event_file(eeg_file_id, original_file_data.path, derivatives)
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/eeg.py", line 637, in fetch_and_insert_event_file
    physiological.insert_event_file(
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/physiological.py", line 551, in insert_event_file
    self.db.insert(
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/database.py", line 193, in insert
    raise Exception("Insert query failure: " + format(err))
Exception: Insert query failure: (1366, "Incorrect integer value: 'None' for column `loris_rb`.`physiological_task_event`.`EventValue` at row 1")
```

Reply:
Good catch, it's easy to solve.

Reply:
@cmadjar Fixed.